File: dataset/output-bk/MISTRAL_codestral-latest/Task110_MISTRAL_codestral-latest.cpp
#include <iostream>
#include <regex>

bool isValidURL(const std::string &url) {
    const std::regex urlPattern("^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]");
    return std::regex_match(url, urlPattern);
}

int main() {
    std::string url = "http://www.example.com";
    if (isValidURL(url)) {
        std::cout << "URL is valid" << std::endl;
    } else {
        std::cout << "URL is not valid" << std::endl;
    }
    return 0;
}

File: dataset/output-bk/MISTRAL_codestral-latest/Task110_MISTRAL_codestral-latest.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class URLValidator {
    public static boolean isValidURL(String url) {
        String urlPattern = "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]";
        Pattern pattern = Pattern.compile(urlPattern);
        Matcher matcher = pattern.matcher(url);
        return matcher.matches();
    }

    public static void main(String[] args) {
        String url = "http://www.example.com";
        if (isValidURL(url)) {
            System.out.println("URL is valid");
        } else {
            System.out.println("URL is not valid");
        }
    }
}